contracts/Router.sol

```diff
@@ -0,0 +1,250 @@
+pragma solidity ^0.6.6;
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+
+
+interface I_ERC20 {
+    function balanceOf(address account) external view returns (uint256);
+    function transfer(address recipient, uint256 amount) external returns (bool);
+    function transferFrom(address sender, address recipient, uint256 amount) external returns (bool);
+    function increaseAllowance(address spender, uint256 addedAmount) public virtual returns (bool);
+}
+
+interface I_TREERewards {
+  function notifyRewardAmount(uint256 reward) external;
+}
+
+
+contract Router {
+    using SafeMath for uint256;
+
+    event Pledge(address addr, uint256 amount);
+    event Unpledge(address addr, uint256 amount);
+    event Rebase(treeSold, reserveTokenReceived);
+    event WithdrawToken(address token, address to, uint256 amount);
+    event SetReserveToken(address token);
+    event SetCharityCut(uint256 _newValue);
+    event SetRewardsCut(uint256 _newValue);
+    event SetGov(address _newValue);
+    event SetCharity(address _newValue);
+    event SetLPRewards(address _newValue);
+
+    address constant private TREE = 0xCE222993A7E4818E0D12BC56376c5a60f92A5783;
+    address constant private RESERVE = 0x390a8Fb3fCFF0bB0fCf1F91c7E36db9c53165d17;
+    address constant private DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+
+    // Constants are from the original Reserve
+    /// @notice precision for decimal calculations
+    uint256 public constant PRECISION = 10**18;
+    uint256 public constant UNISWAP_GAMMA = 997 * 10**15;
+    /// @notice the minimum value of charityCut
+    uint256 public constant MIN_CHARITY_CUT = 10**17; // 10%
+    /// @notice the maximum value of charityCut
+    uint256 public constant MAX_CHARITY_CUT = 5 * 10**17; // 50%
+    /// @notice the minimum value of rewardsCut
+    uint256 public constant MIN_REWARDS_CUT = 5 * 10**15; // 0.5%
+    /// @notice the maximum value of rewardsCut
+    uint256 public constant MAX_REWARDS_CUT = 10**17; // 10%
+
+    address private gov;
+    address private charity;
+    uint256 private charityCut;
+    uint256 private rewardsCut;
+    uint256 private oldReserveBalance;
+    bool private hasTransferredOldReserveBalance;
+
+    I_ERC20 public tree = I_ERC20(TREE);
+    I_ERC20 public reserveToken = I_ERC20(DAI);
+    I_TREERewards public lpRewards;
+
+    uint256 private totalPledged;
+    uint256 private numPledgers;
+    uint256 private treeSold;
+    mapping (uint256 => address) private pledgers;
+    mapping (address => uint256) private amountsPledged;
+ 
+    constructor(address _gov, address _charity, address _lpRewards, uint256 _charityCut, uint256 _rewardsCut, unit256 _oldReserveBalance) public {
```

Comment:
`oldReserveBalance` type is also misspelled here

Reply:
Seems to be fixed now